htdocs/web_portal/controllers/downtime/downtime_utils.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+/*_____________________________________________________________________________
+ *=============================================================================
+ * File: downtime_utils.php
+ * Description: Helper functions which can be re-used while adding
+ *              or editing a downtime.
+ *
+ * License information
+ *
+ * Copyright 2023 STFC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ /*====================================================== */
+require_once __DIR__ . '/../../../../lib/Gocdb_Services/Factory.php';
+
+use DateTime;
+use DateTimeZone;
+
+/**
+ * Sorts the impacted IDs into impacted services and impacted endpoints.
+ *
+ * @param array $impactedIDs An array of `impactedIDs` which user has selected.
+ *
+ * @return array An array containing
+ *               `$siteLevelDetails` and `serviceWithEndpoints`.
+ */
+function endpointToServiceMapping($impactedIDs)
+{
+    $siteLevelDetails = [];
+    $serviceWithEndpoints = [];
+
+    /**
+     * For each impacted ID,
+     * sort between endpoints and services using the prepended letter.
+     */
+    foreach ($impactedIDs as $id) {
+        /**
+         * `$siteNumber` => It's about Site ID
+         * `$parentService` => It's about service ID endpoint belongs too
+         * `idType` => It's about to differentiate
+         *             the endpoint vs service selection.
+         */
+        list($siteNumber, $parentService, $idType) = explode(':', $id);
+
+        $type = strpos($idType, 's') !== false ? 'services' : 'endpoints';
```

Comment:
I'd go for something like:

```suggestion
    foreach ($impactedIDs as $id) {
        list($siteId, $parentServiceId, $idType) = explode(':', $id);

        // put typical input for $idType here. 
        $type = strpos($idType, 's') !== false ? 'services' : 'endpoints';
```